### PR TITLE
Improve verify goal

### PR DESCRIPTION
**Pull request by albers**
The current logic behind the verify goal is insufficient.

It scans **all** directories in the `resultsFolder` for failed assertions. If the `resultsFolder` contains reports from previous failed executions, this will produce false positives.

The recommended workaround for this situation was to always call `mvn clean`.

With this PR, the `GatlingMojo` creates a file in `resultsFolder` with the names of directories that were created during Gatling's execution. The `VerifyMojo` ignores all directories that are not contained in this file.

### CODE REVIEW

**Comment by slandelle** [comment]
I suspect you're breaking the current behavior the Jenkins plugin uses.

**Comment by albers** [comment]
The only change in GatlingMojo is that it writes an additional file. The JUnit files are still processed for **all** directories. Should be safe.

File: src/main/java/io/gatling/mojo/GatlingMojo.java [hunk 1]
```diff
@@ -229,10 +237,17 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
     } finally {
-        copyJUnitReports();
+      recordSimulationResults();
     }
   }
 
+  private Set<File> directoriesInResultsFolder() {
+    File[] directories = resultsFolder.listFiles(File::isDirectory);
+    return (directories == null)
+            ? new HashSet<>()
```
Comment (slandelle) [inline src/main/java/io/gatling/mojo/GatlingMojo.java]: Collections.emptySet()

File: src/main/java/io/gatling/mojo/GatlingMojo.java [hunk 2]
```diff
@@ -286,18 +301,41 @@ private void executeGatling(List<String> gatlingJvmArgs, List<String> gatlingArg
...
+      for (File directory : directoriesInResultsFolder()) {
+        if (isNewDirectory(directory)) {
+          writer.write(directory.getName() + "\n");
+        }
+      }
+    }
+  }
+
+  private boolean isNewDirectory(File directory) {
+    return ! existingDirectories.contains(directory);
```
Comment (slandelle) [inline src/main/java/io/gatling/mojo/GatlingMojo.java]: format (remove extra space)

**Comment by albers** [comment]
Thanks for your review. I just pushed an updated version. Please let me know if there are more issues.

**Comment by albers** [comment]
@slandelle I'm planning to create one more verification related PR soon. It will make the `VerifyMojo` respect the `gatling.skip` property.

**Comment by slandelle** [comment]
@albers Thanks!

> I'm planning to create one more verification related PR soon. It will make the VerifyMojo respect the gatling.skip property.

It would be great. Please note that we'll most likely release Gatling 3.1.0 at this end of this month, so it would be great to then release a new version of the maven plugin too.

**Pull request merged**